source/simulator/core.cc

```diff
@@ -1976,6 +1974,23 @@ namespace aspect
           break;
         }
 
+        case NonlinearSolver::Binary_input:
+        {
+          AssertThrow (!parameters.free_surface_enabled,
+                       ExcMessage (std::string("Currently, we do not support free surface mesh with the \"nonlinear\" binary input scheme.")));
+
+          std::string filename = parameters.binary_data_directory + "/" + parameters.binary_data_file_name + Utilities::int_to_string(timestep_number, 5) + ".mesh";
+
+          triangulation.load(filename.c_str());
```

Comment:
The mesh is written into a binary file using Boost in the first pass at each time step via the velocity binary plugin. Granted that we read each of these files at the appropriate time step, the meshes will be identical down to the bits. As an added bonus, I believe that we can use this model when AMR is enabled during the first pass.